ddpui/ddpairbyte/airbyte_service.py

```
@@ -498,593 +505,500 @@ def get_source_schema_catalog(
 
     res = abreq("jobs/get_without_logs", {"id": job_id})
+    if "job" not in res:
+        error_message = f"Job not found: {job_id}"
+        logger.error(error_message)
+        raise HttpError(404, error_message)
+
     return res
 
 
 def get_jobs_for_connection(
     connection_id: str, limit: int = 1, offset: int = 0, job_types: list[str] = ["sync"]
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)